### DISCUSSION TITLE
VS Code Vibrancy

### DISCUSSION THREAD

**Discussion by jrkmtf** [ramensoftware/windhawk-mods#28] [Q&A]
There is a plugin for VS Code called Vibrancy, which applies a translucent/Acrylic effect to the VS Code by editing its css file. Seems like the VSCode Tweaker Windhawk mod should be able to achieve the same effect, but I'm having trouble getting it to work. I copied and saved the Vibrancy css code, added a "Code from a file" code snippet in Windhawk, and pointed it to this .css file, but it did not seem to have any effect.

Any thoughts on how to achieve a similar effect through this Windhawk mod?

**Comment by m417z** [comment]
The reason that doesn't work is that the Windhawk mod affects the following two files:

* vs/workbench/workbench.desktop.main.js
* vs/workbench/workbench.desktop.main.css

While VS Code Vibrancy [modifies the following files](https://github.com/EYHN/vscode-vibrancy/blob/d2f3c95a7b8e1afd09b0ee1d196c8bf164c5e591/extension/index.js#L115-L116), in addition to copying its own files inside the VSCode folder:

* vs/code/electron-browser/workbench/workbench.html
* main.js

Also, the modification of workbench.html is not a simple code append, it's a modification of an existing meta tag inside the file.

The reason for all this is likely to be able to use Electron APIs early on and load native code, while the VSCode Tweaker Windhawk mod was designed for plain JS and CSS customizations.

Perhaps @EYHN, the author of VS Code Vibrancy, would like to port VS Code Vibrancy to be a Windhawk mod. it shouldn't be difficult. Or I can do it if the authors don't mind.

Alternatively, I can extend the VSCode Tweaker mod to support the two additional files above and to support replacement in addition to append. In this case, you'll still have to copy the VS Code Vibrancy folder manually and configure the code modifications. Having a VS Code Vibrancy Windhawk mod which installs in one click can be nice and provide a better user experience.

**Comment by m417z** [comment]
I made a couple of changes and released VSCode Tweaker v1.0.1. Now you can modify the target files, and there's an option to search and replace, so you should be able to replicate what VS Code Vibrancy does. Let me know how it goes or if you need help.
